### ISSUE TITLE
Pre-Release Refactor #1 (2020.1.0-α)

### ISSUE THREAD

**Issue by BOZ-CODE** [BOZ-CODE/tomi#26]
There's probably going to be one other pre-release refactor before officially deploying an alpha release of the _Threat Observance and Mitigation Interface_ to a production server. A lot of developments have happened so far with 2020.1.0-α and now that I have a general idea of what I'm working with (to some extent), I feel like it's best to clean up and refactor a lot of the pre-existing code. So here are a few things that have happened so far:
- An initial development of the _engine system_ which will handle the _heart of TOMI_ (so to speak).
- An initial development of the _informatics system_ which will handle the _soul of TOMI_ (so to speak).
- An initial development of the _network system_ which will handle how _TOMI_ communicates with the outside world.

I feel the way I will be implementing features throughout these systems will be incredibly subjective. The _engine system_, for example, should be able to accept some type of communication from the _network system_ which is then processed via the _informatics system_. Afterwards, the _engine system_ should be able to then display the result of some type of outcome from the _informatics system_. Another way of thinking about the tri-system setup is that the _engine system_ will be responsible for the overall system-wide configuration and filesystem. The _informatics system_ will be responsible for information processing (which will heavily rely on the database as well, I'm sure). And the _network system_ is responsible for specific ways that _TOMI_ can connect to the outside world (i.e. [Discord](https://discord.com/) for now).

Once this first major refactoring is finished, my next plan is to begin implementing the very thing that _TOMI_ was designed **(see: should be)** to handle: threat monitoring and mitigation. This will **(see: should)** utilize all three major systems.

A lot of time was also spent just getting used to configuring a development server, deploying _TOMI_ to the development server, and just dealing with the overall headache of bugs and unexpected issues that come with starting up a new project. I believe that I now have a good understanding of what I'll be dealing with now after these past few days of coding, but I'm sure I'll need one more major refactor before the alpha release will be ready for deployment.

**Comment by BOZ-CODE** [comment]
[_Progress so far._](https://github.com/BOZ-CODE/tomi/compare/cf29f3f3c87c%5E...09bc19a3a3a8)

Worked on a better system to determine what internal environment the software may be in as well as being able to identify the operating system. ***Mode.java*** and ***Environment.java*** are responsible for this. If things go wrong, you might run into one of the two exception classes I've created: ***UnknownEnvironmentException.java*** and ***UnsupportedEnvironmentException.java***. If this happens, the software will terminate early on. Created two test classes as well though they weren't exactly necessary: ***ModeTest.java*** and ***EnvironmentTest.java***.

**TL; DR** - If you run the program in a production environment, it will likely fail if your operating system is not a type of Linux. If you run the software in a development environment, you can run it on one of the four operating systems listed inside a private enum class within the ***Environment.java*** source file. This assumes you're planning on locally testing the software... using a development server is a different situation altogether and completely up to you if you want to fiddle with the source code enough to have it operate on any kind of operating system (that isn't Linux-based).

**Comment by BOZ-CODE** [comment]
First major refactor for the pre-release is finished per this [_closed pull request_](https://github.com/BOZ-CODE/tomi/pull/27).
